src/exo/API_cursors.py

```diff
@@ -1005,6 +1011,38 @@ def lift_cursor(impl, proc):
         assert False, f"bad case: {type(impl)}"
 
 
+def find(scope: C, proc: API.Procedure, pattern: str, many: bool):
+    """
+    Find the most specific possible cursor for the given pattern in
+    the given scope of the proc. For example, a pattern matching a
+    single assignment statement will return an AssignCursor, not a
+    StmtCursor or BlockCursor.
+
+    If the optional parameter `many` is set to True, then return a list,
+    potentially containing more than one Cursor.
+
+    In any event, if no matches are found, a SchedulingError is raised.
+    """
+    if not isinstance(pattern, str):
+        raise TypeError("expected a pattern string")
+    default_match_no = None if many else 0
+    raw_cursors = match_pattern(
+        scope, pattern, call_depth=1, default_match_no=default_match_no
+    )
+    assert isinstance(raw_cursors, list)
+    cursors = []
+    for c in raw_cursors:
+        c = lift_cursor(c, proc)
+        if isinstance(c, (BlockCursor, ExprListCursor)) and len(c) == 1:
+            c = c[0]
```

Comment:
This line looks a bit suspicious, isn't this saying that you'll return the first statement of the block if `.find` matches with multiple statements? Can you add a test for querying and scoping with block cursors?

Reply:
That's not what this is doing. Each element of `raw_cursor` is a match. This is more of an artifact of how `pattern_match` interacts with the internal `BlockCursor`. When we match a statement `s` in a body or some expr in a list of accesses (e.g. `i` in `x[i, j, i]`), `match_pattern` returns a list (e.g. `[s]` or `[i]`). This code is to essentially unwrap that before returning it to the user.

Reply:
I see that makes sense